### PR TITLE
Document Dockerization; project JSON files

### PR DESCRIPTION
**Pull request by kitcambridge**
@rpappalax, @pdehaan, @darkwing What do you think?

### CODE REVIEW

File: JSON.rst [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+<http://www.thekelleys.org.uk/dnsmasq/doc.html>`_, records the start and stop
+time, and launches the step container with a set of environment variables and
+command-line arguments. Heka sends CPU, memory, and statsd metrics collected
+from running instances to a central InfluxDB node for analysis.
+
+A step contains the following properties:
+
+* ``name`` (String): The step name.
+* ``instance_count`` (Integer): The desired number of instances for this step.
+* `instance_region` (String): The `EC2 region
```
Comment (pdehaan) [inline JSON.rst]: Should these be double backticks since this is an .rst file?

Comment (kitcambridge) [inline JSON.rst]: Yes. :stuck_out_tongue_winking_eye: Thanks for catching that!

**Comment by rpappalax** [comment]
thanks, Kit!  looks great.  We can always update it later if anything's unclear.

**Comment by pdehaan** [comment]
Travis failed... r-

File: MOZILLA.rst [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+#. Click the "Upload" button to open the file uploader, then add the
+   ``{project}.tar.bz2`` file.
+#. Click "Set Details >", then "Set Permissions >". You can skip the details
+   screen; there's no need to enable redundant storage or encryption.
+#. Click "Add more permissions" and select the "Everyone" grantee from the
+   drop-down. Check the "Open/Download" box to allow public downloads.
+#. Click "Start Upload".
+#. Once the upload completes, the tarball will be accessible at
+   ``https://s3.amazonaws.com/loads-docker-images/{project}.tar.bz2``. You
+   can specify this URL in the project file's `container_url` field.
```
Comment (pdehaan) [inline MOZILLA.rst]: This should be \`\`container_url\`\`

File: JSON.rst [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  be mounted as read-write. Paths are subject to interpolation.
+* ``docker_series`` (String, optional): The InfluxDB time series containing CPU
+  and memory stats for this step. Defaults to ``"stats".``
+* ``prune_running`` (Boolean, optional): Whether unresponsive running instances
+  should be terminated. Defaults to ``true``.
+
+Interpolation
+=============
+
+In addition to the variables specified in the `environment_data` field, Loads
```
Comment (pdehaan) [inline JSON.rst]: This should be \`\`environment_data\`\`.

File: JSON.rst [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  to the container ``ENTRYPOINT``, or the full command name and arguments if
+  the container's ``Dockerfile`` does not specify an entry point. Subject to
+  interpolation.
+* ``dns_name`` (String, optional): A round-robin DNS name for all instances in
+  this step. For example, if an application in a step starts an HTTP server on
+  port 8000, setting the ``dns_name`` to ``test.mozilla.dev`` allows testers
+  in subsequent steps to make requests to ``http://test.mozilla.dev:8000``.
+  This is useful for creating test clusters.
+* ``port_mapping`` (Comma-separated string, optional): A mapping of container
+  ports to host ports, in the form of `container:host`.
```
Comment (pdehaan) [inline JSON.rst]: This should be \`\`container:host\`\`.

**Comment by pdehaan** [comment]
Sorry, three last nits. Otherwise, hopefully Travis will be green next commit and we can merge.
Thanks @kitcambridge! :cookie:

**Comment by kitcambridge** [comment]
Travis failures...

**Pull request merged**